dashboard/app/models/script.rb

```diff
@@ -998,14 +1000,31 @@ def self.add_script(options, raw_stages, new_suffix: nil, editor_experiment: nil
       end
       # Set/create Stage containing custom ScriptLevel
       if stage_name
+        # check if that lesson_group exists for the script
+        lesson_group = nil
+        if lesson_group_name
+          puts script.lesson_groups
+          lesson_group = script.lesson_groups.detect {|lg| lg.name == lesson_group_name} ||
```

Comment:
Do we enforce that lesson group names are unique anywhere?

Reply:
Would we want/need that? Two different scripts should be able to each have a lesson groups with the same name and within a script if two stages have the same lesson group they reference the same lesson group.

Reply:
I updated this line because I realized based on your other comments we don't need the detect. So now its just the find_or_create. Does that help here?

Reply:
Yea I think this is good for now; I was thinking about the hypothetical case where there are two lesson groups in the DB under the same script, although there's probably no way to create that through normal means.

Dave's comments in https://github.com/code-dot-org/code-dot-org/pull/34042 talk about changing the unique identifier to a different ID than the name, so it seems like we'll have to revisit this if / when we do that, but that would cover this more solidly. I think ideally whatever we're relying on as the unique ID is enforced to be unique in the DB and is indexed.